examples/RNOneSignal/redux-index.js

```diff
@@ -0,0 +1,156 @@
+/**
+ * Sample React Native App
+ * https://github.com/facebook/react-native
+ * @flow
+ */
+
+/**
+
+    This example assumes there's a parent Component that returns the redux Provider
+    wrapping its children with the same store as being imported here being passed
+    as the store prop.
+
+**/
+
+import React, { Component } from "react";
+import { connect } from "react-redux";
+import { AppRegistry, AsyncStorage, StyleSheet } from "react-native";
+import { persistStore } from "redux-persist"; // redux-persist@4.10.1 -- the API for ^5.x.x and rehydrating/purging the store is slightly different
+import OneSignal from "react-native-onesignal";
+import App from "./src/App";
+import { store } from "./configureStore.js"; // Whereever you're creating and exporting your store from
+
+// Action creator -- likely to be in the directory with the rest of your project's action creators
+// placed here for the sake of example
+const setInitialNotification = openResult => ({
+  type: T.INITIAL_NOTIFICATION,
+  payload: openResult.notification.payload.additionalData
+});
+
+// function to initially add onOpened listener to
+// handles initial tap of push notification from an unopened AppState
+const handleOnOpened = openResult => {
+  console.log(`** INITIAL OPEN RESULT FROM APP.JS **`, openResult);
+  console.log("Message: ", openResult.notification.payload.body);
+  console.log("Data: ", openResult.notification.payload.additionalData);
+  console.log("isActive: ", openResult.notification.isAppInFocus);
+  console.log("openResult: ", openResult);
+
+  store.dispatch(setInitialNotification(openResult));
+};
+
+OneSignal.configure({});
+OneSignal.addEventListener("opened", handleOnOpened);
+
+class UnconnectedRNOneSignal extends Component {
+  state = {
+    rehydrated: false
+  };
+
+  componentWillMount() {
+    OneSignal.addEventListener("received", this.onReceived);
+    OneSignal.addEventListener("registered", this.onRegistered);
+    OneSignal.addEventListener("ids", this.onIds);
+
+    this.rehydrateStore();
+  }
+
+  componentWillUnmount() {
+    OneSignal.removeEventListener("received", this.onReceived);
+    OneSignal.removeEventListener("opened", this.onOpened);
```

Comment:
Is this remove right? I believe the function registered for `opened` is `handleOnOpened`, correct me if I'm wrong.

Reply:
@GuiCavi `handleOnOpened` is only registered to the `opened` event when the file is first loaded, before the component mounts, so as to catch any push notification that was tapped with the app in a `closed` state (not `background`).

Once the component mounts and the redux store is rehydrated (or however you “rehydrate” your app’s data), `handleOnOpened` is removed and `this.onOpened` is added.

I don’t believe this workaround is still necessary, however, when using `react-native-onesignal`.

Reply:
Oh, I missed L81 and L82 in this file. Sorry for missing it. Btw, why is this not necessary anymore? I was told to implement it as a prevention to unexpected behavior (that I never see myself). 

Thanks for answering 😄

Reply:
No problem, my friend!

You’ve answered your own question, really lol. The unexpected behavior — you never see it, and I, personally, haven’t seen it recently, either.

Since I made the PR for this workaround, `react-native-onesignal` has been updated multiple times. I believe one of the updates addressed catching an `opened` event within a class that gets mounted as (usually) a child to the Parent that is initially mounted on app load.

When I whipped up this workaround, I put it above my class in my root `index.js` so it was called before the `Root` component’s class instantiation.

Reply:
Thanks for explaining. I'll try to change to a traditional handling and see how it goes.

🤗